File: Diary/Models/Entities/BaseEntity.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Dairy.Entities
{
    public abstract class BaseEntity
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public DateTime? CreatedAt { get; set; }
        public DateTime? DeletedAt { get; set; }
    }
}

File: src/entities/BaseEntity.ts
import { Column, PrimaryGeneratedColumn } from 'typeorm';

export default abstract class BaseEntity {
    
    @PrimaryGeneratedColumn ()
    id: number;

    @Column({
        nullable: true
    })
    createdAt: Date;

    @Column({
        nullable: true
    })
    deletedAt: Date;
}